### PR TITLE
[4.0] We don't really need javascript for tooltips

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
This was discussed in JAB and there was an agreement that JS is not needed for tooltips. Reasons:
- css can do that (cheeper, less code, less executable code)
- it's still accessible

Drawbacks:
- No more html elements in the tooltips

### Testing Instructions
Apply the patch and check the global configuration page (field: Use URL Rewriting)

### Expected result
Tooltips still rendering as before, but without jQuery and bootstrap dependency or any other js code 

### Actual result

screen shot 2018-05-21 at 16 17 53

### Documentation Changes Required
This PR also removes the archaic Mootools way of creating tooltips (class `hasTip`) since moo tools is not with us anymore 😓

@wilsonge 
@mbabker 
@laoneo 

@ciar4n you can take it from here...

### CODE REVIEW

File: libraries/cms/html/uitips.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		{
+			// Pass texts through JText if required.
+			if ($translate)
+			{
+				$content = Text::_($content);
+			}
+
+			$result = str_replace(
+				[
+					'</br>',
```
Comment (Quy) [inline libraries/cms/html/uitips.php]: This too `<br />`?

File: libraries/cms/html/uitips.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	 * @param   string   $title      Unused, kept for B/C
+	 * @param   string   $content    The content to tooltip.
+	 * @param   boolean  $translate  If true will pass texts through JText.
+	 * @param   boolean  $escape     If true will pass texts through htmlspecialchars.
+	 *
+	 * @return  string  The tooltip string
+	 *
+	 * @since   4.0.0
+	 */
+	public static function tipText($title = null, $content = '', $translate = true, $escape = true) {
```
Comment (Quy) [inline libraries/cms/html/uitips.php]: Move `{` to its own line.

File: libraries/cms/html/uitips.php (hunk 1 above)
Comment (dgrammatiko) [reply libraries/cms/html/uitips.php]: maybe we can preg_reg and delete all the html elements...

File: libraries/cms/html/uitips.php (hunk 2 above)
Comment (dgrammatiko) [reply libraries/cms/html/uitips.php]: done

**Comment by mbabker** [comment]
> No more html elements in the tooltips

As long as we still support something like Bootstrap's Popover component (which really should be the one you're using HTML with, we really hacked away at tooltips to make it behave like the popover component in 3.x) this isn't an issue.

**Comment by simbus82** [comment]
I would prefer to have the possibility to insert html in the tooltips. We often develop complex sites with Joomla where we create complex combinations of additional fields, some of which are also used to compose tooltips that must contain HTML.

**Comment by dgrammatiko** [comment]
> We often develop complex sites with Joomla where we create complex combinations of additional fields

Fix your forms. From a UX perspective I will quote Elon musk: If it needs manual it's already broken:
elon-musk-quote-about-any-product-1

**Comment by mbabker** [comment]
@simbus82 For your use case that is where an extended JavaScript component like Bootstrap's Popover component comes into play (and should be what you are using).  Though they may have similar uses, tooltips should be decoupled from any JavaScript tooling by default (and the native browser implementation does not use HTML, that is only possible through JavaScript enhancement).

Which is why I posted my original comment, we do need to support the use case you're looking for, BUT, that should be a separate tool and NOT the Joomla default behavior for tooltips.

**Comment by dgrammatiko** [comment]
@mbabker I guess the special case for popup is when a title is present: https://github.com/joomla/joomla-cms/pull/20519/files#diff-51258b89e12460b1a7f367c2ca89de6eR31

If so we need to fire a different behaviour (but how do we make this pluggable?)

**Comment by simbus82** [comment]
@dgrammatiko When you have to work with Joomla to develop interfaces for intranet applications, used by non-qualified personnel, you will see that what Elon Musk says, no matter how beautiful and true it is, is no longer valid. :-P
But as @mbabker said, we can use the Bootstrap 4 Popover in future.
Thanks to all

**Comment by dgrammatiko** [comment]
@simbus82 nope: https://docs.joomla.org/Special:MyLanguage/Note_form_field_type

**Comment by Bakual** [comment]
@dgrammatiko No need to criticize the work from another person where you neither know the product nor the person. Please stop that.
Also as a general rule, generalised statements are always wrong 😄

**Pull request closed without merging**